### PR TITLE
Fix crash due division by zero in validated operator expression

### LINKED ISSUE #59556: GDScript 2.0: Divide/Modulo by 0 cause the game crash without report in output/console
### Godot version

4.0 alpha 5

### System information

Windows 10 / Kubuntu

### Issue description

In some case, I accident produce `0` that would be used in module and then it would cause the silent crash. In windows console, nothing. In Linux console, stack trace log are print. ~(too lazy to reboot to linux again so no log provide)~ The editor output/debugger also show nothing from both OS.
I expect that it will show something like `Division By Zero in operator '%'.` in debugger after it force pause the game like what Godot 3 does.

### Steps to reproduce

This code:
```GDScript
var zero: int = 0
print(1 % zero)
print(1 / zero)
```

### Minimal reproduction project

_No response_

### LINKED ISSUE #62765: GDScript 2.0: 0/int(variable) ignores zero division errors
### Godot version

4.0alpha11

### System information

Windows 10, GLES3,  NVIDIA Quadro P2000

### Issue description

Using `0/int(variable)` appears to make zero division errors stop working even if the variable is zero. As a result, the game crashes without warning.

```
extends Node2D

var test = 0
var test2 = 0
# Called when the node enters the scene tree for the first time.
func _ready() -> void:
	# This results in an error "Division by zero error in operator '/'."
#	test2 = 0/test
	# This causes crash without displaying an error.
	test2 = 0/int(test)
```

### Steps to reproduce

1. Make a new project 
2. Make one node in the tree(probably its class doesn't matter)
3. Copy the code above, attach it to the node and run the project

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by Chaosus**
- Fix https://github.com/godotengine/godot/issues/62765

*Bugsquad edit: This closes https://github.com/godotengine/godot/issues/59556.*

### CODE REVIEW

File: modules/gdscript/gdscript_vm.cpp [hunk 1]
```diff
@@ -688,7 +689,17 @@ Variant GDScriptFunction::call(GDScriptInstance *p_instance, const Variant **p_a
...
+
+				if (!valid) {
+					if ((*dst).get_type() == Variant::STRING) {
+						err_text = (*dst);
+						err_text += " in operator '" + Variant::get_operator_name((Variant::Operator)operator_idx) + "'.";
+					} else {
+						err_text = "Invalid operands '" + Variant::get_type_name(a->get_type()) + "' and '" + Variant::get_type_name(b->get_type()) + "' in operator '" + Variant::get_operator_name((Variant::Operator)operator_idx) + "'.";
+					}
+					OPCODE_BREAK;
+				}
```
Comment (cdemirer) [inline modules/gdscript/gdscript_vm.cpp:702]: Should this also be inside `#ifdef DEBUG_ENABLED` like `OPCODE_OPERATOR` (just above this one)?

File: core/variant/variant_op.h [hunk 2]
```diff
@@ -266,7 +272,7 @@ class OperatorEvaluatorModNZ {
 		*r_ret = a % b;
 		r_valid = true;
 	}
-	static inline void validated_evaluate(const Variant *left, const Variant *right, Variant *r_ret) {
+	static inline void validated_evaluate(const Variant *left, const Variant *right, Variant *r_ret, bool &r_valid) {
 		*VariantGetInternalPtr<R>::get_ptr(r_ret) = *VariantGetInternalPtr<A>::get_ptr(left) % *VariantGetInternalPtr<B>::get_ptr(right);
 	}
```
Comment (cdemirer) [inline core/variant/variant_op.h:277]: ModNZ is probably also affected. Also there seems to be other template specializations for Vector3i etc.

File: modules/gdscript/gdscript_vm.cpp (hunk 1 above)
Comment (Chaosus) [reply modules/gdscript/gdscript_vm.cpp]: I guess...

**Comment by vnen** [comment]
The idea for validated functions was to avoid the checks, but since it only validates the types, it does makes sense to validate the values at runtime to avoid crashes. So the idea looks okay to me, we just need to check if @reduz agrees with the change.

Then it needs to fix the failing tests and memory leaks found by the CI.

**Pull request closed without merging**